test/promise/evil-promises.js

```diff
@@ -6,7 +6,16 @@ describe('Evil promises should not be able to break invariants', function () {
   specify('resolving to a promise that calls onFulfilled twice', function (done) {
     // note that we have to create a trivial subclass, as otherwise the
     // Promise.resolve(evilPromise) is just the identity function.
-    var EvilPromise = function (executor) { Promise.call(this, executor); };
+    // (And in fact, most native Promise implementations use a private
+    // [[PromiseConstructor]] field in `Promise.resolve` which can't be
+    // easily patched in an ES5 engine, so instead of
+    // `Promise.resolve(evilPromise)` we'll use
+    // `new Promise(function(r){r(evilPromise);})` below.)
+    var EvilPromise = function (executor) {
+      var self = new Promise(executor);
+      Object.setPrototypeOf(self, EvilPromise.prototype);
```

Comment:
Please make sure this test bails out early if `Object.setPrototypeOf` is not shimmed in the current environment.